bip-rpob-tx-template.mediawiki

```diff
@@ -0,0 +1,284 @@
+                                        <77-byte-rpob_signature_message>,
+                                        <16-byte-rpob_hash_tag>
+                                        )
+
+rpob_hash_tag:                16-bytes, utf-8-hash_tag ("rpob/BTC/MAINNET")
+                                    or, utf-8-hash_tag ("rpob/BTC/TESTNET")
+
+rpob_signature_message:       77-bytes, < 1-byte-version_flags>
+                                        <32-byte-input_txid_hash>
+                                        < 4-byte-input_index>
+                                        < 8-byte-burn_output_value>
+                                        <32-byte-rpob_revocation_puzzle>
+</pre>
+
+*The <code>rpob_schnorr_signature</code> uses the [https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki BIP-340] algorithm.
+
+==Deployment==
+
+While the ''IsStandard'' rules are quite restricting, it is quite possible to submit transactions directly to miners, or co-operate with miners who would enjoy to have some addition fee-revenue. So the initial process of testing on the main-network should be possible.
+
+If this standard gains significant attention, we are happy to write a supplementary BIP to define a new service bit to allow nodes to signal that this new type of standard transaction is accepted.
+
+==Rationale==
+===Why require RPoB transactions to use a new transaction version?===
+
+This allows for future indexing and enforcement strategies at the consensus layer.
+
+===Why use a hash puzzle for the revocation?===
+
+The hash-puzzle allows for easy indexing of the revocation, in the format:
+
+<code><32-byte-revocation-puzzle> (if revoked: <32-byte-revocation-puzzle-preimage>)</code>
+
+*Since the digest algorithm is double sha256, even for a very large number of revocation, this index would be very cheap to verify.
+
+*We envision nodes keeping track of all the revocation-puzzles listed in the blockchain, as 32-byte each, a million RPoB is only 32mb. This can be further optimised.
+
+Additionally, we do not want to confuse a public key, (that can sign), and a revocation-puzzle (that may only revoke).
+
+===Why must the revocation output be of zero value?===
+
+Revocation can be spent by anyone once the revocation pre-image has been published.
+
+===Why does a RPoB need a signature at all?===
+
+The primary purpose of the signature is to stop replay-attacks. (Somebody takes your public-key and makes transaction where you don't control the revocation).
+
+The secondary purpose is to allow for untrusted parties to create transactions on your behalf. i.e. a 'RPoB' transaction creation service.
+===Why waste precious block space?===
+
+Consuming block-space is a of secondary proof-of-burn that this proposal takes advantage of, as itself is limited and contested for (blocks are almost full on average). There is an opportunity cost of the next best transaction when a miner chooses to include a RPoB transaction their block.
```

Comment:
This sounds like an attack on Bitcoin.

Reply:
We didn't quite comprehend what attack you we specially referencing. However, we took some time to expand on this point to make our intentions and philosophy more clear:

--

> The authors of this BIP do accept the premise of this question. Each block is a auction of block-space that helps rewarding the miner in their important consensus role. We are not some authority who can define 'waste' from 'economic' transaction. We take the neutral position that if a transaction is willing to pay the transaction-fee-at-auction it must have a economic purpose that supports doing so.
>
> Additionally, this proposal takes advantage of economic opportunity cost that the miner faces when excluding transaction when blocks are full (that they generally are). This allows for the transaction-fees-at-auction to behave as a secondary proof-of-burn.
>
> The authors are not fazed by the possibility that proposal, if gaining significant adoption, will lead to the a meaningful increase of the Bitcoin transaction fees. We regard transaction fees important for long-term health of the network and consider making statements in global consensus networks as inherently costly.